### ISSUE TITLE
LuaTable.length should throw an error but it doesn't

### ISSUE THREAD

**Issue by Zamiell** [TypeScriptToLua/TypeScriptToLua#1076]
Using it as a property should throw an error and tell the end-user to use it as a method instead.

```ts
const test = new LuaTable();

print(test.length);
```

**Comment by Perryvw** [comment]
There is a check to prevent misuse of unary operator mappings, but it is not always present, and probably on the wrong AST nodes:

```ts
const test = new LuaTable<number, number>();

// Expected error is here, but on 'n' instead of 'test.length';
const n = test.length;

// No error on property access
print(test.length);
```

[Playground link](http://localhost:3000/play#code/MYewdgzgLgBFCm0YF4ZngdxgGQK4EMAVfAIwBt4AeMXAWxPgCcAaNOhxgPgAoBKAbgBQggPQiYAUQAeAB3jAEAExhNGIRjACWEGAAsm8ViVyxwMAORhzWyAnzKQAMwsJoAOgpgA5lF3mhoLZoKHCIUB7w3r5CouIAciAqjGoaZjJqcoxQAJ4w+MDAiBCC6ZpgUNyu4Z4+ugJAA)
